bot/cogs/grades_cog/grades_cog.py

```diff
@@ -121,45 +190,92 @@ async def grades(self, ctx, course: str, year: int=MIN_YEAR):
         """
     )
     @ext.short_help('Provides info about a given professor')
-    @ext.example(['prof brian dean'])
-    async def prof(self, ctx, *, prof: str):
-        prof = prof.lower()
-        normalized_name = prof.title()
-
-        if not self.grades_df.Instructor.str.contains(prof).any():
+    @ext.example(('prof brian dean', 'prof kristi whitehead honors'))
+    async def prof(self, ctx, *args):
+        honors = ""
+        if args[-1] in ("honors", "hon"):
+            honors = "honors"
+        elif args[-1] in ("non-honors", "non-hon", "regular", "normal"):
+            honors = "non-honors"
+        elif args[-1] == "all":
+            honors = "all"
+
+        if honors:
+            # User specified correct honors attribute; prof name is all but last word
+            prof = " ".join(args[:-1])
+        else:
+            # No honors argument (or incorrect), assume prof name is entire string
+            prof = " ".join(args)
+
+        if not self.grades_df.Instructor.str.contains(prof, case=False).any():
             embed = discord.Embed(title="Professors", color=Colors.Error)
             result = f'"{prof}" is not a known Professor\n'
             embed.add_field(name="ERROR: Professor doesn't exist", value=result, inline=False)
-            embed.add_field(name=f'Help:', value= f'Run `{await self.bot.current_prefix(ctx)}prof list` to find what professors are available', inline=False)
+            embed.add_field(name=f'Help:',
+                            value=f'Run `{await self.bot.current_prefix(ctx)}prof list` to find what professors are available',
+                            inline=False)
             return await ctx.send(embed=embed)
 
-        #check for if there is a 0% A rate, that means it was a pass fail class and we dont handle those
-        df = self.grades_df[(self.grades_df.Instructor == prof) & (self.grades_df.A > 0)]
+        # check for if there is a 0% A rate, that means it was a pass fail class and we dont handle those
+        df = self.grades_df[(self.grades_df.Instructor.str.lower() == prof.lower()) & (self.grades_df.A > 0)]
```

Comment:
we dont need to lower the names of the profs, its done in the normalize script, and lowering of the arg should be done at the top

Reply:
I had it lower the instructor name to sort of future proof in case we design future CSV files to maintain the capitalization of the prof names. I was thinking this would be preferable to using .title() in case some profs have abnormal capitalization in their names.